doc/en/fixture.rst

```diff
@@ -149,442 +627,966 @@ marked ``smtp_connection`` fixture function.  Running the test looks like this:
 
 .. code-block:: pytest
 
-    $ pytest test_smtpsimple.py
+    $ pytest test_module.py
     =========================== test session starts ============================
     platform linux -- Python 3.x.y, pytest-6.x.y, py-1.x.y, pluggy-0.x.y
     cachedir: $PYTHON_PREFIX/.pytest_cache
     rootdir: $REGENDOC_TMPDIR
-    collected 1 item
+    collected 2 items
+
+    test_module.py FF                                                    [100%]
+
+    ================================= FAILURES =================================
+    ________________________________ test_ehlo _________________________________
+
+    smtp_connection = <smtplib.SMTP object at 0xdeadbeef>
+
+        def test_ehlo(smtp_connection):
+            response, msg = smtp_connection.ehlo()
+            assert response == 250
+            assert b"smtp.gmail.com" in msg
+    >       assert 0  # for demo purposes
+    E       assert 0
+
+    test_module.py:7: AssertionError
+    ________________________________ test_noop _________________________________
+
+    smtp_connection = <smtplib.SMTP object at 0xdeadbeef>
+
+        def test_noop(smtp_connection):
+            response, msg = smtp_connection.noop()
+            assert response == 250
+    >       assert 0  # for demo purposes
+    E       assert 0
+
+    test_module.py:13: AssertionError
+    ========================= short test summary info ==========================
+    FAILED test_module.py::test_ehlo - assert 0
+    FAILED test_module.py::test_noop - assert 0
+    ============================ 2 failed in 0.12s =============================
+
+You see the two ``assert 0`` failing and more importantly you can also see
+that the **exactly same** ``smtp_connection`` object was passed into the
+two test functions because pytest shows the incoming argument values in the
+traceback.  As a result, the two test functions using ``smtp_connection`` run
+as quick as a single one because they reuse the same instance.
+
+If you decide that you rather want to have a session-scoped ``smtp_connection``
+instance, you can simply declare it:
+
+.. code-block:: python
+
+    @pytest.fixture(scope="session")
+    def smtp_connection():
+        # the returned fixture value will be shared for
+        # all tests requesting it
+        ...
+
+
+Fixture scopes
+^^^^^^^^^^^^^^
+
+Fixtures are created when first requested by a test, and are destroyed based on their ``scope``:
+
+* ``function``: the default scope, the fixture is destroyed at the end of the test.
+* ``class``: the fixture is destroyed during teardown of the last test in the class.
+* ``module``: the fixture is destroyed during teardown of the last test in the module.
+* ``package``: the fixture is destroyed during teardown of the last test in the package.
+* ``session``: the fixture is destroyed at the end of the test session.
+
+.. note::
+
+    Pytest only caches one instance of a fixture at a time, which
```

Comment:
Hmm does this mean parametrized fixtures effectively negate scope?

Reply:
In some sense, I'd say so. The scope still acts as a bubble around the fixtures inside it, but parameterizing creates additional bubbles within that it, one for each parameter set. Sort of like sub-scopes.

I thought I added a bit to clarify how that works, but looks like I didn't. I'll put in some stuff for this today.